util/packages.go

```diff
@@ -74,6 +74,12 @@ func getPackagePaths(allPaths []string) ([]string, error) {
 			if info.IsDir() {
 				log.Printf("%-20s\t%10s\t%s", dirs[0], dirs[1], path)
 				foundPaths = append(foundPaths, path)
+			} else {
+				// Not expected file, return nil in order to continue processing sibling directories
+				// Fixes an annoying problem when the .DS_Store file is left behind and the package
+				// is not loading without any error information
+				log.Printf("error: unexpected file: %s", path)
+				return nil
```

Comment:
I wonder if we should return an error instead. This is at first inconvenient for the developer working with the package because he has to fix it but it assures no "invalid" files are inside the package.

@mtojek @ycombinator Do we have any validation in place in `elastic-package check` that would detect these files?

Reply:
I'm wondering how did @aleksmaus run into this issue? Have you been using the elastic-package to boot up the stack? Did you interact with OSX file browser?

@ruflin This directory is not part of a package (look at the path). It's a bug in package-registry. Speaking about validation, there is such one in place, but there are places in which it's not a bug but a feature, like e.g. a package root. It's hard to detect these files as the end-user may include additional READMEs, license notes, etc. 

For reference: https://github.com/elastic/package-spec/blob/master/versions/1/spec.yml#L8 

@aleksmaus I don't think we need to fail the entire application in case of noticing strange files. It will be more convenient to the end-user to skip this files. Could you try this piece of code?

```go
if info.IsDir() {
    log.Printf("%-20s\t%10s\t%s", dirs[0], dirs[1], path)
    foundPaths = append(foundPaths, path)
    return filepath.SkipDir
}
return nil
```

Reply:
Mac OS finder creates these pesky .DS_Store files for Finder app. 
This change doesn't fail the whole app, it keeps loading the package versions. 
Before the change, stumbling on such file (not dir code path) was returning filepath.SkipDir which skips all the siblings directories. 
In my case the directory layout was 
```
.DS_Store
0.1.0
0.1.2
```

And the existing code was stumbling upon .DS_Store file first and since it was not directory the default code path was always returning filepath.SkipDir

From documentation of filepath.Walk: 
```
 If the function returns SkipDir when invoked on a non-directory file, Walk skips the remaining files in the containing directory.
 ```
 
 Sure can update the code  the sample you provided, it's pretty much the same what my code does.
 Would still like to keep the log statement pointing at the error during load.